fix build issues

coderbyheart opened the pull request:
This fixes multiple issues that prevented the CI from being able to build.

Most notably: 

1. the pricing JSON has a property named `$` which breaks the TypeScript types generated by graphQL codegen: these keys were replaced with `USD`.
2. the new libraries include to render the globe depend on browser globals like `window` and `ResizeObserver` which is not available when rendering the page on the server. These components are now lazy loaded in the browser only.

coderbyheart committed:
ci: wait for dev server to be up

This makes the action wait until the dev server is actually up,
not just 30 seconds

ghost commented:
<details>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=483&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org">Click on the image</a> for a new way to code review</b>
</summary>

  - Make big changes easier — review code in small groups of related files
  - Know where to start — see the whole change at a glance
  - Take a code tour — explore the change with an interactive tour
  - Make comments and review — all fully sync’ed with github

    [Try it now!](https://app.codesee.io/r/reviews?pr=483&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org)
</details>

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/distributeaid/distributeaid.org/483/b7f2e95d/f0c8920aa5f7c0ca5ece04b12e4ba07b5ec4735f.svg)](https://app.codesee.io/r/reviews?pr=483&src=https%3A%2F%2Fgithub.com%2Fdistributeaid%2Fdistributeaid.org)

#### Legend
![CodeSee Map Legend](https://app.codesee.io/Legend_v2.png)

coderbyheart committed:
ci: print server.log

coderbyheart committed:
ci: set timeout to 5 minutes

coderbyheart committed:
fix: replace $ in JSON with USD

The "$" key used in the JSON resulted in
graphql-schema-generator creating invalid
TypeScript definitions.

coderbyheart committed:
style: format source

coderbyheart committed:
fix: only tail

coderbyheart committed:
fix: handle undefined values

coderbyheart committed:
fix: use Node.js 16

String.replaceAll is not available in Node.js 14

coderbyheart committed:
fix: add missing definition

coderbyheart committed:
fix: only render globe in browser

The library has a dependency to window

coderbyheart committed:
fix: wrap component

coderbyheart committed:
fix: wrap more components
